### PR TITLE
Add memory intensive R5 instances

### PR DESCRIPTION
**Pull request by huydhn**
ExecuTorch is requesting memory intensive R5 instances because they uses lots of memory to export the model while CPU cores stay idle.  Atm, they need to jump from `2xlarge` to `4xlarge` or `12xlarge` (or even `24xlarge`) to get the necessary amount of memory, which is a waste.

I also allocate more disk space to there instances, enough that they could dump the whole memory without running out of space.

cc @guangy10 @dvorjackz

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: .github/scale-config.yml [hunk 1]
```diff
@@ -220,3 +220,27 @@ runner_types:
     instance_type: g5.4xlarge
     is_ephemeral: false
     os: windows
+  linux.2xlarge.memory:
+    disk_size: 200
+    instance_type: r5.2xlarge
+    is_ephemeral: false
+    os: linux
+    ami: al2023-ami-2023.5.202*-kernel-6.1-x86_64
```
Comment (guangy10) [inline .github/scale-config.yml:228]: What is the size of the RAM for a memory intensive runner?

Comment (huydhn) [reply .github/scale-config.yml:228]: Here you go https://aws.amazon.com/ec2/instance-types/r5/ (comparing with the current c5 instances you are using https://aws.amazon.com/ec2/instance-types/c5/), let me also draft a post to announce this to a broader audience with these information.

**Review by guangy10 — COMMENTED** [review]
@huydhn This is lovely! Thank you for supporting this asking for ExecuTorch! We should start replacing expensive runners to these memory intensive runners in ExecuTorch CI to balance the cost. cc: @kirklandsign @mergennachin @dbort

File: .github/scale-config.yml (hunk 1 above)
Comment (guangy10) [reply .github/scale-config.yml:228]: Yes, please do post this new runner instance. In ExecuTorch we should start replacing the expensive runners with this new type of runner for most use-cases, and should also guide the developers to always try with this memory intensive runner first before going with a 12x runner or so (to check the workload is memory bound or cpu bound)

File: .github/scale-config.yml [hunk 2]
```diff
@@ -220,3 +220,27 @@ runner_types:
...
+  linux.8xlarge.memory:
+    disk_size: 400
+    instance_type: r5.8xlarge
+    is_ephemeral: false
+    os: linux
+    ami: al2023-ami-2023.5.202*-kernel-6.1-x86_64
+  linux.12xlarge.memory:
+    disk_size: 600
```
Comment (dvorjackz) [inline .github/scale-config.yml:242]: Thank you 🙏🏻

**Pull request merged**